class-00/lab-b/starter-code/app.js

```diff
@@ -165,36 +160,37 @@ Student.prototype.greeting = function() {
 
 // TODO: Uncomment the following line of code to see the output in the browser console
 // Note that the arrow function will cause this method to break!
-// console.log(joe.greeting());
+console.log(joe.greeting());
+//Williams Osunkwo: I turned this function to an arrow function, and per the comment above it did break, so i changed it back to a function expression so i could continue on the other functions below without having the error.
 
-
-Student.courseName = function() {
-  return 'This student is enrolled in Code 301.';
-};
+Student.courseName = () => 'This student is enrolled in Code 301.';
 
 // TODO: Uncomment the following line of code to see the output in the browser console
-// console.log(Student.courseName());
+console.log(Student.courseName());
 
 
 
 // STEP 11
 // How do arrow functions affect constructor functions?
-Student.prototype.scope = function() {
+Student.prototype.scope = () => {
   console.log(this);
 };
 
 // TODO: Uncomment the following line of code to see the output in the browser console
-// console.log(joe.scope());
+console.log(joe.scope());
+
 
 Student.prototype.scopeArrow = () => console.log(this);
 
 // TODO: Uncomment the following line of code to see the output in the browser console
-// console.log(joe.scopeArrow());
+console.log(joe.scopeArrow());
 
 // TODO: Write a COMMENT below to answer the following questions.
 // 1. What is "this" when joe.scope() is invoked?
-//
+// It outputted this = Window {postMessage: ƒ, blur: ƒ, focus: ƒ, close: ƒ, parent: Window, …}. I have no idea what that is, but it looks big
 // 2. What is "this" when joe.scopeArrow() is invoked?
-//
+
+// It outputted this = Window {postMessage: ƒ, blur: ƒ, focus: ƒ, close: ƒ, parent: Window, …}. I have no idea what that is, but it looks big
+
 // 3. Explain why "this" is different when an arrow function is used.
-//
+// Because arrow functions dont work well with constructors, and it causes "This" to become globally scoped

```

Comment:
You are mostly correct, but not 100% correct. Arrow functions do behave differently when it comes to scoping contextual 'this.'